pkg/provisioner/ironic/ironic.go

```diff
@@ -465,8 +465,38 @@ func getCPUDetails(cpudata *introspection.CPUType) metal3v1alpha1.CPU {
 	return cpu
 }
 
+func getFirmwareDetails(firmwaredata introspection.ExtraHardwareDataSection) metal3v1alpha1.Firmware {
+
+	// handle bios optionally
+	var bios metal3v1alpha1.BIOS
+
+	if _, ok := firmwaredata["bios"]; ok {
+
+		// we do not know if all fields will be supplied
+		// as this is not a structured response
+		// so we must handle each field conditionally
+		if _, ok := firmwaredata["bios"]["vendor"]; ok {
+			bios.Vendor = firmwaredata["bios"]["vendor"].(string)
```

Comment:
This will panic if the data contains something other than a string. I don't think that can happen in practice today, but it'd be safer to handle that case. I'll submit a follow-up PR.